src/Kmd.Logic.Gateway.Automation/Publish.cs

```diff
@@ -203,6 +203,34 @@ private async Task CreateOrUpdateApis(Guid subscriptionId, Guid providerId, IEnu
             }
         }
 
+        private async Task UpdateApi(IGatewayClient client, Guid subscriptionId, ApiValidationResult apiVersionValidationResult, string folderPath, IList<GetProductListModel> allProducts, ApiVersion apiVersion)
+        {
+            var productIds = apiVersion.ProductNames.Select(n => allProducts.SingleOrDefault(p => string.Compare(p.Name, n, comparisonType: StringComparison.OrdinalIgnoreCase) == 0)?.Id)?.ToList();
```

Comment:
May be we can take this to calling method since this is required for create/update. One line of code will be reduced

Reply:
Will take care next PR  hope this is fine